component-server-parent/component-server/src/main/java/org/talend/sdk/component/server/service/IconResolver.java

```diff
@@ -104,38 +119,59 @@ public Icon resolve(final Container container, final String icon) {
             }
         }
         final ClassLoader appLoader = Thread.currentThread().getContextClassLoader();
-        return cache.icons
-                .computeIfAbsent(icon,
-                        k -> ofNullable(getOverridenIcon(icon, appLoader)
-                                .orElseGet(() -> doLoad(container.getLoader(), icon)
-                                        .orElseGet(() -> doLoad(appLoader, icon).orElse(null)))))
-                .orElse(null);
+        if (isThemeSupported) {
+            final String appliedTheme = theme != null ? theme : defaultTheme;
+            final String themedIcon = appliedTheme + "/" + icon;
+            Icon themed = cache.icons
+                    .computeIfAbsent(icon + appliedTheme,
```

Comment:
Can we use a "themedIcon" as a key in this Map?

Reply:
yep.